### ISSUE TITLE
Fedora Support?

### ISSUE THREAD

**Issue by massatt212** [mcgillij/amdfan#24]
can you support fedora, i love this program since it works on my rx 6400 compare to amdgpu-fan

**Comment by mcgillij** [comment]
I don't have any fedora boxes to test it out on, however it should be pretty straight forward to pull down the repo and load the systemd.service as long as you have these dependencies installed: https://github.com/mcgillij/amdfan/blob/0dbff2eeca9d7f5f634d7e8bfc2dcb0aa5d4d7e0/PKGBUILD#L10

**Comment by mcgillij** [comment]
https://github.com/mcgillij/amdfan/blob/main/src/amdfan/amdfan.service here's the systemd service file, I just have it symlinked to `/etc/systemd/system/multi-user.target.wants` similar things should work fine in fedora.

**Comment by massatt212** [comment]
So basically it's a copy paste job and activate service?

**Comment by mcgillij** [comment]
Yeah, there's nothing special going on here, it's just a python script with a systemd service to start it up.

**Comment by l33m4n** [comment]
> https://github.com/mcgillij/amdfan/blob/main/src/amdfan/amdfan.service here's the systemd service file, I just have it symlinked to `/etc/systemd/system/multi-user.target.wants` similar things should work fine in fedora.

This did not work for me. I just had to remake my systemd file so thanks for the great starting point. Perhaps it's because i installed via python. this is my systemd file 

``  GNU nano 7.2                            /usr/lib/systemd/system/amdfan.service                                      
[Unit]
Description=amdfan controller
After=multi-user.target
Requires=multi-user.target

[Service]
ExecStart=/usr/local/bin/amdfan --daemon
Restart=always

[Install]
WantedBy=final.target



only slightly different. but works for me

**Comment by mcgillij** [comment]
> > https://github.com/mcgillij/amdfan/blob/main/src/amdfan/amdfan.service here's the systemd service file, I just have it symlinked to `/etc/systemd/system/multi-user.target.wants` similar things should work fine in fedora.
> 
> This did not work for me. I just had to remake my systemd file so thanks for the great starting point. Perhaps it's because i installed via python. this is my systemd file
> 
> `` ``` GNU nano 7.2 /usr/lib/systemd/system/amdfan.service [Unit] Description=amdfan controller After=multi-user.target Requires=multi-user.target
> 
> [Service] ExecStart=/usr/local/bin/amdfan --daemon Restart=always
> 
> [Install] WantedBy=final.target
> 
> ```
> 
> 
> 
> only slightly different. but works for me 
> 
> idk how im messing up the markdown so badly but i think u guys get it 
> ```

Yeah, the paths will be different based on where you installed it/how (if you grabbed it from pypi or github etc).

**Comment by l33m4n** [comment]
> > > https://github.com/mcgillij/amdfan/blob/main/src/amdfan/amdfan.service here's the systemd service file, I just have it symlinked to `/etc/systemd/system/multi-user.target.wants` similar things should work fine in fedora.
> > 
> > 
> > This did not work for me. I just had to remake my systemd file so thanks for the great starting point. Perhaps it's because i installed via python. this is my systemd file
> > `` ``` GNU nano 7.2 /usr/lib/systemd/system/amdfan.service [Unit] Description=amdfan controller After=multi-user.target Requires=multi-user.target
> > [Service] ExecStart=/usr/local/bin/amdfan --daemon Restart=always
> > [Install] WantedBy=final.target
> > ```
> > 
> > 
> > 
> > only slightly different. but works for me 
> > 
> > idk how im messing up the markdown so badly but i think u guys get it 
> > ```
> 
> Yeah, the paths will be different based on where you installed it/how (if you grabbed it from pypi or github etc).

i gotchya, figured it was good practice to post here for reference of other users.

**Comment by massatt212** [comment]
i cant get it to work at all on fedora

**Comment by l33m4n** [comment]
> i cant get it to work at all on fedora

What problems are you experiencing with the program not working? Do you have any error messages you can report? I have used this program on Fedora 37-38 Xfce spin, as well as currently Fedora 38.

**Comment by massatt212** [comment]
let me reinstall it again, i see u have pip support, how do i install it via pip

**Comment by l33m4n** [comment]
> let me reinstall it again, i see u have pip support, how do i install it via pip

I know this isn't good practice so I'd like to let mcgillij answer this for you. But I personally run 
`sudo pip install amdfan`
I don't know if this is correct because to my knowledge you shouldn't install pip programs as root. Although amdfan requires root permission. So again, mcgillij's answer will be much better. I just do this because it works for me.

**Comment by massatt212** [comment]
× amdfan.service - amdfan controller
     Loaded: loaded (/usr/lib/systemd/system/amdfan.service; enabled; preset: d>
     Active: failed (Result: exit-code) since Sat 2023-05-27 15:26:16 AST; 2s a>
   Duration: 674us
    Process: 5862 ExecStart=/usr/bin/amdfan --daemon (code=exited, status=203/E>
   Main PID: 5862 (code=exited, status=203/EXEC)
        CPU: 729us

May 27 15:26:16 sscstv systemd[1]: amdfan.service: Scheduled restart job, resta>
May 27 15:26:16 sscstv systemd[1]: Stopped amdfan.service - amdfan controller.
May 27 15:26:16 sscstv systemd[1]: amdfan.service: Start request repeated too q>
May 27 15:26:16 sscstv systemd[1]: amdfan.service: Failed with result 'exit-cod>
May 27 15:26:16 sscstv systemd[1]: Failed to start amdfan.service - amdfan cont>
lines 1-13/13 (END)

**Comment by l33m4n** [comment]
> × amdfan.service - amdfan controller Loaded: loaded (/usr/lib/systemd/system/amdfan.service; enabled; preset: d> Active: failed (Result: exit-code) since Sat 2023-05-27 15:26:16 AST; 2s a> Duration: 674us Process: 5862 ExecStart=/usr/bin/amdfan --daemon (code=exited, status=203/E> Main PID: 5862 (code=exited, status=203/EXEC) CPU: 729us
> 
> May 27 15:26:16 sscstv systemd[1]: amdfan.service: Scheduled restart job, resta> May 27 15:26:16 sscstv systemd[1]: Stopped amdfan.service - amdfan controller. May 27 15:26:16 sscstv systemd[1]: amdfan.service: Start request repeated too q> May 27 15:26:16 sscstv systemd[1]: amdfan.service: Failed with result 'exit-cod> May 27 15:26:16 sscstv systemd[1]: Failed to start amdfan.service - amdfan cont> lines 1-13/13 (END)

Can i see ur systemd file pls 
`/usr/lib/systemd/system/amdfan.service` to be exact

**Comment by massatt212** [comment]
[Unit]
Description=amdfan controller
After=multi-user.target
Requires=multi-user.target

[Service]
ExecStart=/usr/bin/amdfan --daemon
Restart=always

[Install]
WantedBy=final.target

**Comment by l33m4n** [comment]
> [Unit] Description=amdfan controller After=multi-user.target Requires=multi-user.target
> 
> [Service] ExecStart=/usr/bin/amdfan --daemon Restart=always
> 
> [Install] WantedBy=final.target

try this and see if it works? If not ur gonna have to wait til the dev can help u.

Unit]
Description=amdfan controller

[Service]
ExecStart=/usr/bin/amdfan --daemon
Restart=always

[Install]
WantedBy=multi-user.target

**Comment by massatt212** [comment]
i think its working, ill go test it soon

**Comment by massatt212** [comment]
did not work, getting 80deg on apex, before i was getting 69deg and the fan rpm on 0 when idle

**Comment by massatt212** [comment]
[Unit]
Description=amdfan controller

[Service]
ExecStart=/usr/local/bin/amdfan --daemon
Restart=always

[Install]
WantedBy=multi-user.target

only arch pkgs put files in the bin folder

**Comment by l33m4n** [comment]
> did not work, getting 80deg on apex, before i was getting 69deg and the fan rpm on 0 when idle

I would wait for the dev to reply then. 


> only arch pkgs put files in the bin folder
`whereis amdfan
amdfan: /usr/local/bin/amdfan`

**Comment by massatt212** [comment]
Ubuntu fedora installs stuff a lil different adds and extra folder usr/local/bin

Arch does usr/bin

**Comment by l33m4n** [comment]
> Ubuntu fedora installs stuff a lil different adds and extra folder usr/local/bin
> 
> Arch does usr/bin

didn't realize you were on ubuntu fedora. Let me spin up a vm. otherwise, again the dev is gonna have to figure it out for u

**Comment by massatt212** [comment]
Ubuntu and Fedora installs some stuff different from arch.

**Comment by mcgillij** [comment]
you can check if it's running with a `systemctl status amdfan`, however your service file should point to the correct location where you installed the script.

**Comment by massatt212** [comment]
But u should write a guide on how to install it on other distros

**Comment by mcgillij** [comment]
> But u should write a guide on how to install it on other distros

I don't use any other distro's sorry, and I made this package for Arch. Feel free to update the wiki or push up a PR with some docs.

**Comment by mcgillij** [comment]
https://github.com/mcgillij/amdfan/wiki I created the GH wiki, so anyone should be able to update it there.

**Comment by l33m4n** [comment]
> https://github.com/mcgillij/amdfan/wiki I created the GH wiki, so anyone should be able to update it there.

i started adding to it but idk how to pr the wiki changes
